composeApp/src/commonMain/kotlin/presentation/screens/feed/FeedScreen.kt

```diff
@@ -0,0 +1,270 @@
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import coil3.compose.AsyncImage
+import org.koin.compose.viewmodel.koinViewModel
+import presentation.NavigationRoutes.Detail
+import presentation.screens.feed.model.Post
+
+@Composable
+fun FeedScreen(
+    navController: NavController,
+    viewModel: FeedViewModel = koinViewModel()
+) {
+    val uiState by viewModel.uiState.collectAsState()
+
+    FeedScreenContent(
+        uiState,
+        openPost = { postId ->
+            navController.navigate("${Detail}/$postId")
+        },
+        onSharePost = { post ->
+            viewModel.onSharePost(post)
+        }
+    )
+}
+
+@Composable
+private fun FeedScreenContent(
+    uiState: FeedUiState,
+    openPost: (String) -> Unit = {},
+    onSharePost: (Post) -> Unit = {}
+) {
+    LazyColumn(
+        modifier = Modifier.fillMaxSize(),
+    ) {
+        items(uiState.posts) {
+            ItemPost(
+                post = it,
+                onClick = {
+                    openPost(it.id)
+                },
+                onShare = {
+                    onSharePost(it)
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun ItemPost(
+    post: Post,
+    onClick: () -> Unit = {},
+    onShare: () -> Unit = {}
+) {
+    PostHeader(post = post)
+    Spacer(modifier = Modifier.height(8.dp))
+    if (post.images.size > 1) {
+        PostMedia(images = post.images) {
+            onClick()
+        }
+    }
+    PostActions(
+        post = post,
+        onShare = onShare
+    )
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun PostHeader(post: Post) {
+    ListItem(
+        text = {
+            Text(
+                text = post.user.name,
+                fontWeight = FontWeight.Medium,
+            )
+        },
+        secondaryText = {
+            Text(
+                text = post.user.username,
+                style = MaterialTheme.typography.body2,
+            )
+        },
+        trailing = {
+            Icon(
+                imageVector = Icons.Outlined.MoreVert,
+                contentDescription = "More options",
+            )
+        },
+        icon = {
+            Box(
+                modifier = Modifier
+                    .size(40.dp)
+                    .clip(CircleShape)
+                    .background(Color.Gray)
+            ) {
+                AsyncImage(
+                    model = post.user.avatar,
+                    contentDescription = "Avatar",
+                    modifier = Modifier.fillMaxSize(),
+                    contentScale = ContentScale.Crop
+                )
+            }
+        }
+    )
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun PostMedia(
```

Comment:
**#suggestion**

We can move this component to be a new file inside of a components package, just to keep the Screen file smaller.